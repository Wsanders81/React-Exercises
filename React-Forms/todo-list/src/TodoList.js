import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import {v4 as uuid} from "uuid"

const TodoList = () => {
	const [ todos, setTodos ] = useState([]);

	const createTodo = (todo) => {
		setTodos((todos) => [ ...todos, todo ]);
	};

	const removeTodo = (id) => {
		setTodos((todos) => todos.filter((t) => t.id !== id));
	};
	return (
		<div>
			<NewTodoForm createTodo={createTodo} />
			{todos.map(({todo, id}) => 
            <Todo text={todo} removeTodo={removeTodo} key={id} id={id}/>)}
		</div>
	);
};

export default TodoList;
