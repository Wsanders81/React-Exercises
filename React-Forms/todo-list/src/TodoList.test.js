import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList'



it('should render without crashing', () => {
    render(<TodoList/>)
})

it('should match snapshot', () => {
    const { asFragment } = render(<TodoList/>)
    expect( asFragment() ).toMatchSnapshot()
})

it('should add new todo', () => {
    const { queryByText, getByLabelText } = render(<TodoList/>)
    const input = getByLabelText("Todo :")
    const btn = queryByText("Submit")
    expect(queryByText('REMOVE')).not.toBeInTheDocument() 
    fireEvent.change(input, {target: {value: "Do Laundry"}})
    fireEvent.click(btn)
    expect(queryByText('REMOVE')).toBeInTheDocument() 
    expect(queryByText('Do Laundry')).toBeInTheDocument() 
 
    
})
it('should delete new todo', () => {
    const { queryByText, getByLabelText } = render(<TodoList/>)
    const input = getByLabelText("Todo :")
    const btn = queryByText("Submit")
    expect(queryByText('REMOVE')).not.toBeInTheDocument() 
    fireEvent.change(input, {target: {value: "Do Laundry"}})
    fireEvent.click(btn)
    const removeBtn = queryByText("REMOVE")
    expect(queryByText('REMOVE')).toBeInTheDocument() 
    expect(queryByText('Do Laundry')).toBeInTheDocument() 
    fireEvent.click(removeBtn)
    expect(queryByText('REMOVE')).not.toBeInTheDocument() 
    expect(queryByText('Do Laundry')).not.toBeInTheDocument() 

    
    
})