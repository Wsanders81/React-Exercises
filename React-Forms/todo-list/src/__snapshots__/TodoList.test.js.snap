// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form
      class="NewTodoForm"
    >
      <label
        for="todo"
      >
        Todo :
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Enter Todo Here"
        type="text"
      />
      <div>
        <button>
          Submit
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
